### ISSUE TITLE
Better handle empty files

### ISSUE THREAD

**Issue by andrew-luhring** [karma-runner/karma#987]
How would one go about clearing karma's cache?
Ex:
      DEBUG [web-server]: serving (cached)

Do I need to do anything special? I'm trying to rule out possibilities as to why I keep getting the "Delaying execution, these browsers are not ready" message. I know I did something last night that started causing this error but I can't remember what- unless it's something in here that I'm missing:

``` js
module.exports = function (config) {
    config.set ({
        basePath      : '../',
        frameworks    : ['mocha', 'requirejs', 'chai'],
        files         : [
            {pattern: 'public/lib/*.js', included: false},
            {pattern: 'public/js/*.js', included: false},
            {pattern: 'tests/_*.js', included: false},
            'tests/test-main.js'
        ],
        exclude       : [
            '**/*min*',
            'public/main.js'
        ],
        client: {
            mocha: {
                    ui: 'bdd'
                //,   reporter : 'nyan'
                ,   growl : true
            }
        },
        reporters     : ['unicorn', 'progress'],
        port          : 9876,
        colors        : true,
        logLevel      : config.LOG_DEBUG,
        autoWatch     : true,
        browsers      : ['Chrome'],
        plugins       : [
            'karma-requirejs',
            'karma-mocha',
            'karma-chai',
            'karma-chrome-launcher',
            'karma-firefox-launcher',
            'karma-unicorn-reporter'
        ],
        captureTimeout: 60000,
        singleRun     : false
    });
};

```

Bonus points if someone can help me figure out how to run Mocha's nyancat reporter. I love that little dude.

**Comment by wilsonpage** [comment]
I rolled Karma back from `v0.12.31` to `v0.12.30` and found this resolved my issue.

I poked around and found that my compiled file that was being watched was being read as empty after the third file change. This was causing Karma to fallback to getting the contents out of its cache ([see here](https://github.com/karma-runner/karma/blob/master/lib/middleware/common.js#L39)).

I don't know enough about Karma internals to take this any further, only that the regression was introduced in `v0.12.31`.
